src/kernels/ipywidgets-message-coordination/ipyWidgetScriptSource.ts

```diff
@@ -87,35 +85,28 @@ export class IPyWidgetScriptSource {
             if (response) {
                 this.uriConverter.resolveUri(response.request, response.response);
             }
+        } else if (message === IPyWidgetMessages.IPyWidgets_Ready) {
+            // Send to UI (even if there's an error) instead of hanging while waiting for a response.
+            (this.scriptProvider ? this.scriptProvider?.getBaseUrl() : Promise.resolve())
+                .then((baseUrl) => {
+                    if (baseUrl) {
+                        this.postEmitter.fire({
+                            message: IPyWidgetMessages.IPyWidgets_BaseUrlResponse,
+                            payload: baseUrl.toString()
+                        });
+                    }
+                })
+                .catch((ex) => {
+                    console.error(`Failed to get baseUrl`, ex);
+                });
+        } else if (message === IPyWidgetMessages.IPyWidgets_IsOnline) {
+            const isOnline = (payload as { isOnline: boolean }).isOnline;
+            this.isWebViewOnline.resolve(isOnline);
```

Comment:
Helps us determine from exension host whether the webview has access to the internet (i.e. whether we can access CDN resources from webview)